persistence-jdbc/core/src/main/scala/com/lightbend/lagom/internal/persistence/jdbc/SlickProvider.scala

```diff
@@ -43,13 +48,21 @@ private[lagom] class SlickProvider(system: ActorSystem)(implicit ec: ExecutionCo
   // users can disable the usage of jndiDbName for userland read-side operations by
   // setting the jndiDbName to null. In which case we fallback to slick.db.
   // slick.db must be defined otherwise the application will fail to start
-  val db = {
+  val db: JdbcBackend.Database = {
     if (readSideConfig.hasPath("slick.jndiDbName")) {
+      // when the DB is obtained via JNDI we trust the creator will handle the shutdown
       new InitialContext()
         .lookup(readSideConfig.getString("slick.jndiDbName"))
         .asInstanceOf[Database]
     } else if (readSideConfig.hasPath("slick.db")) {
-      Database.forConfig("slick.db", readSideConfig)
+      val unmanaged: jdbc.JdbcBackend.Database = Database.forConfig("slick.db", readSideConfig)
```

Comment:
It seems that we were not shutting down the database when configuring using `slick.db`.